engine/eConstr.ml

```diff
@@ -445,10 +445,22 @@ let fold sigma f acc c = match kind sigma c with
 let compare_gen k eq_inst eq_sort eq_constr nargs c1 c2 =
   (c1 == c2) || Constr.compare_head_gen_with k k eq_inst eq_sort eq_constr nargs c1 c2
 
+let eq_einstance sigma i1 i2 =
+  let i1 = EInstance.kind sigma (EInstance.make i1) in
+  let i2 = EInstance.kind sigma (EInstance.make i2) in
+  Univ.Instance.equal i1 i2
+
+let eq_esorts sigma s1 s2 =
+  let s1 = ESorts.kind sigma (ESorts.make s1) in
+  let s2 = ESorts.kind sigma (ESorts.make s2) in
+  Sorts.equal s1 s2
```

Comment:
Shouldn't these live in EInstance/ESorts?

Reply:
Probably, but this makes the boostrap API bigger, so...

Reply:
You can add to esorts in econstr.ml without touching bootstrap, you just have to do a small manipulation to avoid double declaring the module:
~~~ocaml
module ESorts = struct
  include Evd.MiniEConstr.ESorts
  let bla = ()
end

include (Evd.MiniEConstr : module type of Evd.MiniEConstr with module ESorts := ESorts)
~~~